python/ray/data/_internal/execution/legacy_compat.py

```diff
@@ -192,6 +192,24 @@ def _blocks_to_input_buffer(blocks: BlockList, owns_blocks: bool) -> PhysicalOpe
         read_tasks = blocks._tasks
         remote_args = blocks._remote_args
         assert all(isinstance(t, ReadTask) for t in read_tasks), read_tasks
+
+        # Defensively compute the size of the block as the max size reported by the
+        # datasource and the actual read task size. This is to guard against issues
+        # with bad metadata reporting.
+        def cleaned_metadata(read_task):
+            block_meta = read_task.get_metadata()
+            task_size = len(cloudpickle.dumps(read_task))
+            if block_meta.size_bytes is None or task_size > block_meta.size_bytes:
+                if task_size > 100000:
+                    print(
+                        f"WARNING: the read task size ({task_size} bytes) is larger "
+                        "than the reported output size of the task "
+                        f"({block_meta.size_bytes} bytes). This may be a size "
+                        "reporting bug in the datasource being read from."
+                    )
```

Comment:
wondering why we want to reset `block_meta.size_bytes` if `task_size` larger than it?

Reply:
We want the upper bound on size for estimation here, to avoid using too much memory by accident.

Reply:
Got it. Can we define the `100000` as a constant in this file? That would be easy for developers to change it when debugging.